gooddata-pandas/gooddata_pandas/result_convertor.py

```diff
@@ -0,0 +1,315 @@
+            # the dimension id specified on the grand total says from what dimension were
+            # the grand totals calculated (1 for column totals or 0 for row totals);
+            #
+            # the grand totals themselves should, however be placed in the opposite dimension:
+            #
+            # column totals are extra rows at the end of the data
+            # row totals are extra columns at the right 'edge' of the data
+            opposite_dim = 1 if dim_idx == 0 else 0
+
+            if self.grand_totals[opposite_dim] is None:
+                # grand totals not initialized yet; initialize both data and headers by making
+                # a shallow copy from the results
+                self.grand_totals[opposite_dim] = grand_total["data"][:]
+                # TODO: row total measure headers are currently not supported (only aggregation info w/o measure label)
+                #       measure header defs are under ["headerGroups"][>0]
+                self.grand_totals_headers[opposite_dim] = grand_total["dimensionHeaders"][0]["headerGroups"][0][
+                    "headers"
+                ][:]
+            elif paging_dim != opposite_dim:
+                # grand totals are already initialized and the code is paging in the direction that reveals
+                # additional grand total values; append them accordingly; no need to consider total headers:
+                # that is because only the grand total data is subject to paging
+                grand_totals_item = cast(List[_DataArray], self.grand_totals[opposite_dim])
+                if opposite_dim == 0:
+                    # have column totals and paging 'to the right'; totals for the new columns are revealed so
+                    # extend existing data arrays
+                    for total_idx, total_data in enumerate(grand_total["data"]):
+                        grand_totals_item[total_idx].extend(total_data)
+                else:
+                    # have row totals and paging down, keep adding extra rows
+                    grand_totals_item.extend(grand_total["data"])
+
+    def result(self) -> _DataWithHeaders:
+        return _DataWithHeaders(
+            data=self.data,
+            data_headers=(cast(_DataHeaders, self.data_headers[0]), self.data_headers[1]),
+            grand_totals=(self.grand_totals[0], self.grand_totals[1]),
+            grand_total_headers=(self.grand_totals_headers[0], self.grand_totals_headers[1]),
+        )
+
+
+def _extract_all_result_data(response: BareExecutionResponse, page_size: int = _DEFAULT_PAGE_SIZE) -> _DataWithHeaders:
+    """
+    Extracts all data and headers for an execution result. This does page around the execution result to extract
+    everything from the paged API.
+
+    :param response: execution response to work with;
+    :return:
+    """
+    num_dims = len(response.dimensions)
+    offset = [0 for _ in range(num_dims)]
+    limit = [page_size for _ in range(num_dims)]
```

Comment:
Could you please be consistent in the initialization of a list of the same value?

[Here](https://github.com/gooddata/gooddata-python-sdk/blob/9e3c4874637ca72c1aa0d7015308767199f3bd98/gooddata-pandas/gooddata_pandas/result_convertor.py#L289) I see the initialization of a list of Nones using * operator, let's prefer that.

Reply:
fixed in last commit